elastic/beats
# [Bug] fix bug with empty filter values returning no results in system/service

## What does this PR do?

This fixes a bug where if the system/service module is running on an OS with an older version of systemd, **and** a user hasn't configured `pattern_filter`, the metricset will return no results. When using the systemd `ListUnitsFiltered` API call, we don't bypass the pattern_filter in cases where the filter is empty.

## Why is it important?

It's a bug, the metricset won't return any data.

## Checklist




## How to test this PR locally

- Build and Run on CentOS 7, which uses a sufficiently old version of systemd.
- Enable the `system/service` metricset
- make sure `service.pattern_filter` isn't configured.
- Make sure we get data.
